snuba/web/db_query.py

```diff
@@ -374,7 +406,10 @@ def execute_query_with_caching(
     with sentry_sdk.start_span(description="execute", op="db") as span:
         if use_cache:
             key = get_query_cache_key(formatted_query)
-            result = cache.get(key)
+            partition_id = reader.get_cache_partition_id()
+            cache_partition = _get_cache_partition(partition_id)
```

Comment:
since these two functions are always meant to be used together, maybe it's better to put it into its own `get_partitioned_cache(reader: Reader) -> RedisCache` function